### ISSUE TITLE
Add support for caching the results of external link checks

### ISSUE THREAD

**Issue by grollinger** [timaschew/link-checker#7]
We use the link checker in our CI pipeline which runs rather frequently.
Because we're also checking external links, we caused some external sites that were the target of our links to throttle us, causing the link check to sometimes fail randomly.

At the moment we disabled checking external links and we're working on setting up a scheduled pipeline to check them every now and then.

But what we'd really like is the ability to cache the result of an external link check somewhere so that the target site is not requested *every* time the link-checker runs. 

I could imagine something like:
```
link-checker --external-link-cache .link-cache --external-link-interval 1week
```

which would cause the link-checker to only *actually* do a request to the link target iff:
- The link hasn't been checked before (it's not saved in the cache yet) or
- The last check was more than one week ago (according to what's saved in the cache)

**Comment by timaschew** [comment]
Sounds good.  

I guess if `--external-link-interval` is not provided we can default to some value. One week is actually also a good default what do you think?

**Comment by grollinger** [comment]
> I guess if `--external-link-interval` is not provided we can default to some value. One week is actually also a good default what do you think?

I'd tend to agree :smile:

**Comment by timaschew** [comment]
PR is ready: #8 
Feel free to give some feedback.

**Comment by timaschew** [comment]
Published version 1.4.0 to npm and dockerhub
